src/spatial/detail/ArborX_ExpandHalfToFull.hpp

```diff
@@ -34,12 +34,13 @@ void expandHalfToFull(ExecutionSpace const &space, Offsets &offsets,
   Kokkos::parallel_for(
       "ArborX::Experimental::HalfToFull::count",
       Kokkos::RangePolicy(space, 0, n), KOKKOS_LAMBDA(int i) {
+        int local_update = 0;
         for (int j = offsets_orig(i); j < offsets_orig(i + 1); ++j)
         {
-          int const k = indices_orig(j);
-          Kokkos::atomic_inc(&offsets(i));
-          Kokkos::atomic_inc(&offsets(k));
+          ++local_update;
+          Kokkos::atomic_inc(&offsets(indices_orig(j)));
```

Comment:
I'd prefer if we still defined `k` for readability.
```suggestion
          int const k = indices_orig(j);
          Kokkos::atomic_inc(&offsets(k));
```